build.gradle

```diff
@@ -7,14 +7,21 @@ plugins {
     id 'io.spring.dependency-management' version '1.0.11.RELEASE'
     id 'com.github.ben-manes.versions' version '0.39.0'
     id "au.com.dius.pact" version "4.2.10"
+    id 'info.solidsoft.pitest' version '1.5.1'
```

Comment:
I added this to run some permutation testing as I picked up a number of issues as I was going along where I'd missed fields or similar. When I got round to running it there weren't any issues in any of the mapping code but unfortunately it can't handle Lombok generated code (discussion [here](https://groups.google.com/g/pitusers/c/PPPNwV14wzc) as to why) so it's not mutating the fluent builder operators like I'd hoped 😕 Still, with all this refactoring and moving stuff around I think it's worth having as it might pick a couple of things here and there.